src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+        }
+	}
+
+	protected void updateProblemsLabel(List<Object> valuesList) {
+	    
+	    // TODO write code
+	    if (valuesList == null || valuesList.size() == 0) {
+            selectedProblemsLabel.setText("None selected");
+            selectedProblemsLabel.setToolTipText("None selected");
+        } else {
+            
+            selectedProblemsLabel.setText(valuesList.size()+" selected");
+            selectedProblemsLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            
+            selectedProblemList = new ArrayList<Problem>();
+            valuesList.forEach((prob) -> {selectedProblemList.add((Problem)prob);});
+        }
+	}
+
+	protected void updateNosLabel(List<Object> valuesList) {
+
+        if (valuesList == null || valuesList.size() == 0) {
+            selectedNosLabel.setText("None selected");
+            selectedNosLabel.setToolTipText("None selected");
+        } else {
+
+            selectedNosLabel.setText(valuesList.size() + " selected");
+            selectedNosLabel.setToolTipText(Arrays.toString(valuesList.toArray()));
+            submissionSolutionList = new SubmissionSolutionList();
+
+            // update/load selected items into submissionSolutionList
+            valuesList.forEach((subSL) -> {
+                submissionSolutionList.add((SubmissionSampleLocation) subSL);
+            });
+
+        }
+    }
+
+	protected void clearTextArea() {
+		textArea.selectAll();
+		textArea.replaceSelection("");
+		linenumber = 0;
+	}
+
+	void addLineToTextArea(String s) {
+
+		linenumber++;
+		String fmtLineNumber = String.format("%4d", linenumber);
+
+		textArea.insert(fmtLineNumber + " " + hhMMSSformatter.format(new Date()) + " " + s + "\n", 0);
+//		textArea.append(s);
+	}
+
+	protected void selectNewCDP() {
+
+		File file = selectEntry("Select CDP");
+
+		File cdpDir = FileUtilities.findCDPConfigDirectory(file);
+
+		if (cdpDir != null) {
+		    updateCDPDirAndFields(cdpDir.getAbsolutePath());
+		} else {
+
+			int result = FrameUtilities.yesNoCancelDialog(this,
+					file.getAbsoluteFile() + " may not be a CDP directory, continue anyways?", "Select CDP");
+			if (result == JOptionPane.YES_OPTION) {
+	            updateCDPDirAndFields(file.getAbsolutePath());
+			}
+		}
+	}
+
+	/**
+	 * Select yaml file/entry.
+	 * 
+	 * @param dialogTitle
+	 * @return
+	 */
+	protected File selectEntry(String dialogTitle) {
+
+		JFileChooser chooser = new JFileChooser(cdpTextField.getText());
+//        chooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
+		FileFilter filterYAML = new FileNameExtensionFilter("YAML document (*.yaml)", "yaml");
+		chooser.addChoosableFileFilter(filterYAML);
+
+		chooser.setAcceptAllFileFilterUsed(false);
+		chooser.setFileFilter(filterYAML);
+
+		int action = chooser.showOpenDialog(this);
+		chooser.setDialogTitle(dialogTitle);
+
+		switch (action) {
+		case JFileChooser.APPROVE_OPTION:
```

Comment:
Personally, I prefer that the `case` elements inside the body of a `switch` statement are indented, just like the code inside an `if` statement or a method body are indented.  I admit that not everybody needs to necessarily agree with this, but it seems like (once upon a time) we had agreed on this as a "PC2 code formatting standard"...